test/js/utils-test.js

```diff
@@ -327,3 +327,126 @@ test("getRoles on element with abstract role.", function() {
     test("getRoles on element with multiple invalid roles.", multipleRoleTestHelper(['foo', 'fubar', 'bar'], -1));
 
 }());
+
+module("isValidNumber");
+
+test("with integer.", function() {
+    var actual = axs.utils.isValidNumber("10");
+    strictEqual(actual.value, 10, "Integer should be parsed");
+    ok(actual.valid, "Integer should be valid");
+});
+
+test("with leading and trailing whitespace.", function() {
+    var actual = axs.utils.isValidNumber(" 10 ");
+    strictEqual(actual.value, 10, "Integer should be parsed");
+    ok(actual.valid, "Integer should be valid");
+});
+
+test("with zero.", function() {
+    var actual = axs.utils.isValidNumber("0");
+    strictEqual(actual.value, 0, "Zero should be parsed");
+    ok(actual.valid, "Integer should be valid");
+});
+
+test("with leading zero integer.", function() {
+    var actual = axs.utils.isValidNumber("09");
```

Comment:
Yes worth considering - I had to think about that... I decided this is not a general purpose number parser but specific to our a11y needs, usually ARIA attributes in HTML (e.g. `aria-valuenow`) - I couldn't find one that allowed for octal and I doubt there are any screen readers that read "ten" when they encounter "012".

Even `parseInt` no longer treats leading zero as octal (tested in Chrome, Firefox and Safari).
`parseInt("012")` returns decimal 12.
There is a good writeup on it here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Octal_interpretations_with_no_radix

So I think it's safe for us to assume leading zero is not octal - if someone has an octal number they should convert it to decimal before we see it in the a11y tools.